src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -523,10 +524,14 @@ public void shutdown(int timeoutSeconds) {
                 logger.atInfo().setEventType("executor-service-shutdown-initiated").log();
             });
             logger.atInfo().log("Waiting for executors to shutdown");
-            boolean executorTerminated = executorService.awaitTermination(timeoutSeconds,
-                    TimeUnit.SECONDS);
-            boolean scheduledExecutorTerminated = scheduledExecutorService.awaitTermination(timeoutSeconds,
-                    TimeUnit.SECONDS);
+            // when kernel shuts down due to external signal, give some time for executor service to stop so that
+            // threads are interrupted correctly
+            int executorServiceShutdownTimeoutSecond =
+                    timeoutSeconds == -1 ? EXECUTOR_SERVICE_SHUTDOWN_TIMEOUT_SECONDS : timeoutSeconds;
+            boolean executorTerminated =
+                    executorService.awaitTermination(executorServiceShutdownTimeoutSecond, TimeUnit.SECONDS);
+            boolean scheduledExecutorTerminated =
+                    scheduledExecutorService.awaitTermination(executorServiceShutdownTimeoutSecond, TimeUnit.SECONDS);
```

Comment:
Do you want to wait this timeout for each executor? Or should you be computing a remaining time for the second await?